tests/phpunit/tests/html-api/wpHtmlTagProcessor.php

```diff
@@ -461,6 +461,36 @@ public function test_next_tag_should_stop_on_closers_only_when_requested() {
 		$this->assertTrue( $p->is_tag_closer(), 'Indicated a tag closer is a tag opener' );
 	}
 
+	/**
+	 * @ticket 56299
+	 *
+	 * @dataProvider data_rcdata_and_script_tags
+	 * @covers WP_HTML_Tag_Processor::next_tag
+	 * @covers WP_HTML_Tag_Processor::is_tag_closer
+	 *
+	 * @param string $tag_name The name of the tag to test.
+	 */
+	public function test_next_tag_should_stop_on_rcdata_and_script_tag_closers_when_requested( $tag_name ) {
+		$p = new WP_HTML_Tag_Processor( "<$tag_name>abc</$tag_name>" );
+
+		$p->next_tag();
+		$this->assertTrue( $p->next_tag( array( 'tag_closers' => 'visit' ) ), 'Did not find the tag closer' );
+		$this->assertTrue( $p->is_tag_closer(), 'Indicated a tag opener is a tag closer' );
+	}
+
+	/**
+	 * Data provider.
+	 *
+	 * @return string[][].
+	 */
+	public function data_rcdata_and_script_tags() {
+		return array(
+			array( 'script' ),
+			array( 'textarea' ),
+			array( 'title' ),
```

Comment:
```suggestion
			'tag name: script'    => array( 'script' ),
			'tag name: textarea' => array( 'textarea' ),
			'tag name: title'    => array( 'title' ),
```

[Per Core's handbook](https://make.wordpress.org/core/handbook/testing/automated-testing/writing-phpunit-tests/#repetitive-tests), use named data sets in a data provider.

Reply:
the handbook highly recommends it but does not require it. seems quite redundant here don't you think?

Reply:
and one hopes that the recommend-not-require is to allow for taking situational needs like this one into account so that blanket rules don't override normal code quality and maintenance concerns…

Reply:
The reasoning for it is:

* Readability when looking at the test's data set.
* Quickly identifying which data set failed, if it fails.

Let's explore both:

* Readability: The data sets are readable as tags.
* Quickly identify which data set failed:

Forcing a failure on all of the data sets and with the first one having a named data set, here are the results:

```
1) Tests_HtmlApi_wpHtmlTagProcessor::test_next_tag_should_stop_on_rcdata_and_script_tag_closers_when_requested with data set "tag name: script" ('script')
Did not find the tag closer
Failed asserting that true is false.

/var/www/tests/phpunit/tests/html-api/wpHtmlTagProcessor.php:477
phpvfscomposer:///var/www/vendor/phpunit/phpunit/phpunit:97

2) Tests_HtmlApi_wpHtmlTagProcessor::test_next_tag_should_stop_on_rcdata_and_script_tag_closers_when_requested with data set #0 ('textarea')
Did not find the tag closer
Failed asserting that true is false.

/var/www/tests/phpunit/tests/html-api/wpHtmlTagProcessor.php:477
phpvfscomposer:///var/www/vendor/phpunit/phpunit/phpunit:97

3) Tests_HtmlApi_wpHtmlTagProcessor::test_next_tag_should_stop_on_rcdata_and_script_tag_closers_when_requested with data set #1 ('title')
Did not find the tag closer
Failed asserting that true is false.
```

The first one gives this information: `with data set "tag name: script" ('script')`.

and the 2nd one gives this information: `with data set #0 ('textarea')`.

The `#0` is misleading and can contributor confusion thinking it might mean the its element in the set, which it's not. That said, the tag itself is listed next to it. It is reasonable to conclude that though `#0` may be confusing, the tag itself is quickly recognizable to identify which one failed.

Reply:
🤷‍♂️

so are you saying you require this even though the guide you linked as the rulebook only recommends it, or are you suggesting it's your personal preference?

Reply:
@hellofromtonya it's okay to have opinions. I'm not asking you to ignore this; I was asking for the context of what you were recommending, since "[Per Core's handbook](https://make.wordpress.org/core/handbook/testing/automated-testing/writing-phpunit-tests/#repetitive-tests)" somewhat contradicted "use named data sets"

Whereas if the point is to follow the guide then it's unclear that this code isn't following the guide, but if you're preference, we can engage in that conversation without bringing the broader implications of the guide into it.